src/main/java/uk/gov/hmcts/reform/em/orchestrator/automatedbundling/AutomatedCaseUpdater.java

```diff
@@ -0,0 +1,78 @@
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import uk.gov.hmcts.reform.em.orchestrator.automatedbundling.configuration.BundleConfiguration;
+import uk.gov.hmcts.reform.em.orchestrator.automatedbundling.configuration.BundleConfigurationFolder;
+import uk.gov.hmcts.reform.em.orchestrator.automatedbundling.configuration.ConfigurationLoader;
+import uk.gov.hmcts.reform.em.orchestrator.service.caseupdater.CcdCaseUpdater;
+import uk.gov.hmcts.reform.em.orchestrator.service.ccdcallbackhandler.CcdCallbackDto;
+import uk.gov.hmcts.reform.em.orchestrator.service.dto.CcdBoolean;
+import uk.gov.hmcts.reform.em.orchestrator.service.dto.CcdBundleDTO;
+import uk.gov.hmcts.reform.em.orchestrator.service.dto.CcdBundleFolderDTO;
+import uk.gov.hmcts.reform.em.orchestrator.service.dto.CcdValue;
+
+import java.util.Optional;
+
+/**
+ * This class will update add a new bundle to case based on some predefined configuration
+ */
+public class AutomatedCaseUpdater implements CcdCaseUpdater {
+    private static final String CONFIG_FIELD = "bundleConfiguration";
+    private final ConfigurationLoader configurationLoader;
+    private final ObjectMapper jsonMapper;
+
+    public AutomatedCaseUpdater(ConfigurationLoader configurationLoader, ObjectMapper jsonMapper) {
+        this.configurationLoader = configurationLoader;
+        this.jsonMapper = jsonMapper;
+    }
+
+    @Override
+    public boolean handles(CcdCallbackDto ccdCallbackDto) {
+        return ccdCallbackDto.getCaseData().has(CONFIG_FIELD);
+    }
+
+    /**
+     * Load the configuration file then add a new bundle to the case data based on that configuration. If an error occurs
+     * during loading or processing the original case data will be returned with errors
+     */
+    @Override
+    public JsonNode updateCase(CcdCallbackDto ccdCallbackDto) {
+        String configurationName = ccdCallbackDto.getCaseData().get(CONFIG_FIELD).asText();
+        BundleConfiguration configuration = configurationLoader.load(configurationName);
+        Optional<ArrayNode> bundles = ccdCallbackDto.findCaseProperty(ArrayNode.class);
+
+        if (!bundles.isPresent()) {
+            throw new InvalidCaseException("Could not find bundles inside case");
+        }
+
+        this.addNewBundle(configuration, bundles.get());
+
+        return ccdCallbackDto.getCaseData();
+    }
+
+    private void addNewBundle(BundleConfiguration configuration, ArrayNode bundles) {
+        CcdBundleDTO bundle = new CcdBundleDTO();
+        bundle.setTitle(configuration.title);
+        bundle.setHasCoversheets(CcdBoolean.Yes);
+        bundle.setHasTableOfContents(CcdBoolean.Yes);
+        int sortIndex = 0;
+
+        for (BundleConfigurationFolder folder : configuration.folders) {
+            CcdBundleFolderDTO ccdFolder = new CcdBundleFolderDTO();
+            ccdFolder.setName(folder.name);
+            ccdFolder.setSortIndex(sortIndex++);
+
+            bundle.getFolders().add(new CcdValue<>(ccdFolder));
```

Comment:
I'd have built up a full list, and done bundle.setFolders(fullListOfFolders). Good to see you can do this within the for loop :)